tasks/main.yml

```diff
@@ -3,6 +3,9 @@
   apt: pkg=ntp={{ ntp_version }} state=present
 
 - name: Configure NTP
-  template: src=ntp.conf.j2 dest=/etc/ntp.conf
+  template: src={{ item }} dest=/etc/ntp.conf
+  with_first_found:
+    - "ntp.{{ ansible_distribution_release }}.conf.j2"
```

Comment:
The templated string here does not match the filenames, which causes the default template to be used all of the time:

```bash
❯ tree templates
templates
├── ntp.conf.default.j2
├── ntp.conf.trusty.j2
└── ntp.conf.xenial.j2

0 directories, 3 files
```